packages/backend/src/indexers/DiscoveryIndexer.test.ts

```diff
@@ -0,0 +1,110 @@
+import { Logger } from '@l2beat/backend-tools'
+import { DiscoveryConfig, DiscoveryEngine } from '@l2beat/discovery'
+import { ChainId, Hash256 } from '@lz/libs'
+import { expect, mockObject } from 'earl'
+
+import { BlockNumberRepository } from '../peripherals/database/BlockNumberRepository'
+import { DiscoveryRepository } from '../peripherals/database/DiscoveryRepository'
+import { IndexerStateRepository } from '../peripherals/database/IndexerStateRepository'
+import { BlockNumberIndexer } from './BlockNumberIndexer'
+import { DiscoveryIndexer } from './DiscoveryIndexer'
+
+describe(DiscoveryIndexer.name, () => {
+  describe(DiscoveryIndexer.prototype.update.name, () => {
+    it('should run discovery on the latest block', async () => {
+      const config = mockConfig()
+      const discoveryEngine = mockObject<DiscoveryEngine>({
+        discover: async () => [],
+      })
+      const discoveryRepository = mockObject<DiscoveryRepository>({
+        addOrUpdate: async () => true,
+      })
+
+      const disocoveryIndexer = new DiscoveryIndexer(
+        discoveryEngine,
+        config,
+        mockObject<BlockNumberRepository>({
+          findAtOrBefore: async () => 1,
+        }),
+        discoveryRepository,
+        mockObject<IndexerStateRepository>(),
+        Logger.SILENT,
+        mockObject<BlockNumberIndexer>({
+          subscribe: () => {},
+        }),
+      )
+
+      expect(await disocoveryIndexer.update(0, 1)).toEqual(1)
+      expect(discoveryEngine.discover).toHaveBeenCalledTimes(1)
+      expect(discoveryEngine.discover).toHaveBeenNthCalledWith(1, config, 1)
+      expect(discoveryRepository.addOrUpdate).toHaveBeenNthCalledWith(1, {
+        discoveryOutput: {
+          version: 2,
+          name: 'test',
+          configHash: configHash,
```

Comment:
```suggestion
          configHash: config.hash,
```

Can we strip config hash out of module scope and either declare it per case or use it by accessing it by mock config object? Just a suite's dependency purity

Reply:
f2e3641